RELEASE-CHECKLIST.md

```diff
@@ -5,4 +5,7 @@
 1. In the `CHANGELOG.md` file, move everything under "Unreleased" to a new section named `## [vx.y.z] – YYYY-MM-DD`
 1. Make a dryrun like so:
 	1. Ensure you're in the project root dir
-	1. Run: `./scripts/publish.sh --dry-run --allow-dirty`
+	1. Run: `./scripts/publish.sh --dry-run`
```

Comment:
This not point is not useful anymore after your findings but the `Note` below is.

Out-of-scope for this PR: I think we should provide another script to perform dry-run for all the crates or update the existing one to work in dry-run mode too.

Reply:
yeah, hmm, I just don't see how we can make it work, not without some crazy shenanigans. Do you?

Reply:
we can just provide another script that performs alternatively have `CLI option` --dry-run that doesn't perform the complicated stuff such as reading remote version and so on.

```bash
for crate in ${ORDER}; do {
   cd $crate
   cargo publish --dry-run
   cd -
}
```

Reply:
//cc @TarikGul :point_up: see my previous comment.

I think `1. Run: /scripts/publish.sh --dry-run` does more harm than good because it doesn't work but the added NOTE is excellent